app/controllers/health_controller.rb

```diff
@@ -0,0 +1,15 @@
+class HealthController < ApplicationController
+  def workers
+    statuses = worker_health_checker.statuses
+
+    status = statuses.all?(&:healthy?) ? :ok : :internal_error
```

Comment:
a nit: maybe this logic could be in the `WorkerHealthChecker` class itself? `worker_health_checker.public_status`